### PR TITLE
105/tc 778 locking gremlin

### PR DESCRIPTION
**Pull request by emarcey**

### CODE REVIEW

**Review by jcapobianco1 — APPROVED** [review]
nice work. when are we going to open source this thing?

**Comment by emarcey** [comment]
> nice work. when are we going to open source this thing?

🤷‍♀️ 
That sounds like a @pmallela question

**Review by akeating-cbi — APPROVED** [review]

File: gremlin_client.go [hunk 1]
```diff
@@ -116,18 +127,28 @@ func (c *GremlinClient) execWithRetry(ctx context.Context, query string, queryId
...
 			}
+			t := time.Since(acq)
+			fmt.Printf("Acquired lock for key %v after %v\n", queryId, t)
+			queryTimes = append(queryTimes, t)
+			if t > maxTime {
+				maxTime = t
+				fmt.Printf("New longest query: %v\n", maxTime)
+
+			}
+			fmt.Printf("Average query time: %v\n", time.Duration(avgTimeSlice(queryTimes)))
```
Comment (akeating-cbi) [inline gremlin_client.go]: If we have a logger available it might make sense to use that instead of prints

**Review by akeating-cbi — APPROVED** [review]

File: gremlin_client.go (hunk 1 above)
Comment (emarcey) [reply gremlin_client.go]: Those were for testing, I'll drop them

**Pull request merged**